sympy/matrices/expressions/matmul.py

```diff
@@ -36,6 +36,8 @@ def __new__(cls, *args, **kwargs):
         factor, matrices = obj.as_coeff_matrices()
         if check:
             validate(*matrices)
+        if not matrices:
+            return factor
```

Comment:
Is this behavior OK? It makes `MatMul()` (with no args) return `1` (the number). Previously it would return an empty MatMul object, which would be semi-broken (e.g., `MatMul().shape` would give an `IndexError`). 

I think the correct behavior would be to return an identity matrix, but it would need a shape. It's a little weird to me that MatMul handles both matrices and scalar factors.